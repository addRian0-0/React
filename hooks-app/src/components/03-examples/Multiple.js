import React from 'react';
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';
import "../01-useState/counter.css";

export const Multiple = () => {

    const { counter, increment } = useCounter(1);
    const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
    const { author, quote } = !!data && data[0];

    return (
        <div>
            <h1>Breaking Bad</h1>

            {
                loading ? (
                    <div className="alert alert-info text-center">
                        Loading...
                    </div>
                ) : (
                    <blockquote className="blockquote text-right" >
                        <p className="" >{quote}</p>
                        <footer className="blockquote-footer">{author}</footer>
                    </blockquote>
                )
            }

            <button
                onClick={increment}
                className="btn btn-primary" >
                Siguiente quote
            </button>

        </div>
    )
}
