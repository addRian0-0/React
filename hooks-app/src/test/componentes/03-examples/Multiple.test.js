import { renderHook, act } from '@testing-library/react-hooks';
import { shallow } from "enzyme";
import { Multiple } from '../../../components/03-examples/Multiple';
import { useFetch } from '../../../hooks/useFetch';

jest.mock('../../../hooks/useFetch');

describe('pruebas en el mutiple component', () => {
    
    test('debe de mostrarse correctamente', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })

        const wrapper = shallow(<Multiple/>);
        expect(wrapper).toMatchSnapshot();
        
    })

    test('debe de mostrar la informacion', () => {
        
        useFetch.mockReturnValue({
            data: [{
                author: 'addRian0-0',
                quote: 'Quien tenga miedo de morir que no nazca'
            }],
            loading: false,
            error: null
        })

        const wrapper = shallow(<Multiple/>);
        expect(wrapper.find('.alert').exists()).toBe(false);

    })
    

})
