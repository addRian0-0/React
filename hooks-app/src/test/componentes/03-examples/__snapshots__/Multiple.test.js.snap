// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el mutiple component debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Breaking Bad
  </h1>
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
